### PR TITLE
CT-7584 Verifying Postgres SSL certificates

### PR DESCRIPTION
**Pull request by eugene7646**

### CODE REVIEW

**Review by APriestman — APPROVED** [review]

**Review by raman-bt — APPROVED** [review]
Made an editorial comment to define constants. But it does not hold up the approval.

File: bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java [hunk 1]
```diff
@@ -714,11 +714,17 @@ Connection getConnection(String databaseName) throws TskCoreException {
...
-				// ssl=true: enables SSL encryption. 
-				// NonValidatingFactory avoids hostname verification.
-				// sslmode=require: This mode makes the encryption mandatory and also requires the connection to fail if it can�t be encrypted. 
-                // In this mode, the JDBC driver accepts all server certificates.
-				url.append("?ssl=true&sslfactory=org.postgresql.ssl.NonValidatingFactory&sslmode=require");
+				// ssl=true: enables SSL encryption. 				
+				if (info.isSslVerify()) {
+					// DefaultJavaSSLFactory: uses Java's default truststore to validate server certificate.
+					// sslmode=verify-ca: verifies that the server we are connecting to is trusted by CA. 
+					url.append("?ssl=true&sslfactory=org.postgresql.ssl.DefaultJavaSSLFactory&sslmode=verify-ca");
```
[the remark this answers is not in the document]
Comment (eugene7646) [reply bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java:721]: @raman-bt @jayaramcs My first reaction was "oh, good idea!". My second reaction though is that while these URL constants are indeed re-used - they are spread across multiple different files. There are 2 files in this PR, and 3 more files on the CT side that would use these constants. So we would have to find an accessible place where to keep these constants. If we want CT to re-use the constants then we would need to create a new public TSK interface. Alternatively CT can create and use it's own URL constants (this is probably better). Thoughts?

[the remark this answers is not in the document]
Comment (raman-bt) [reply bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java:721]: Separate constants in TSK and CT seems to be the least disruptive choice.

[the remark this answers is not in the document]
Comment (eugene7646) [reply bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java]: @jayaramcs Both PRs have been updated to use constants and are ready to merge

**Pull request merged**